### DISCUSSION TITLE
Votación para política de ramas en testing

### DISCUSSION THREAD

**Discussion by perealberto** [Aparking/AparKing_Backend#116] [Polls]
¡Buenas a todos! He abierto esta votación para decidir cómo actuar a partir de ahora con la ramificación a la hora de hacer tests. Si bien es verdad que algunos compañeros están conformes con los procedimientos actuales, otros tantos creen que es demasiado engorroso tener que crear una nueva rama y prefieren hacer los tests en la de la misma feature. Por tanto, los coordinadores hemos decidido ceder la decisión al equipo. Aquí abajo se describen las alternativas. 

Contexto: un compañero termina de desarrollar su funcionalidad en la rama **feat** y hace merge mediante una pull request a la rama **epic**. Entonces:

- Opción 1 (actual): se crea una rama nueva que tiene como type = 'test' y de nombre el mismo que el de la rama de funcionalidad. Además, se crea una issue de testing de dicha funcionalidad y se asocia la rama a ella.
- Opción 2 (nueva): no se crea ninguna rama nueva. Se mantiene la rama de funcionalidad y se indica que se están haciendo tests porque se ha tenido que crear previamente una issue de testing de dicha funcionalidad, asociando en ella la rama que mantenemos para hacer el testing.

**Comment by MarioArocaPaez** [comment]
¡Epa! Yo estoy a full con mantener la rama 'feature', ¿sabes? Al final de cuentas, es todo sobre hacer esa funcionalidad chévere y, una vez esté lista, el colega se mete a la rama y revisa lo que se cocinó. Además, si los tests pillan algún fallo (que, entre tú y yo, para eso están), es mucho más fácil para el master de la 'feat' entrar y arreglar la movida. Porque, si vamos al caso, creo que no hemos dejado claro qué onda si surge algún meneo: si pegarle una arreglada en la rama original o en la que se armó para los tests. ¡Vaya lío, colega!

**Reply by marvicmar** [reply]
Estoy con Mario, mantener la rama feat creo que va a facilitar más la tarea de testing al compañero ya que se va a encontrar con la funcionalidad aislada y no con todas las características de la épica y diferentes compañeros trabajando en la misma rama realizando test (que ya sabemos que cuando hay una rama conjunta y se mezcla código vienen los problemas de incompatibilidades de unos con otros).

**Reply by PedroRuizAguilar** [reply]
Yo también creo que dejarlo en la feature, de manera aislada es mejor, a parte que uno se ahorra ver tanta nueva rama en el repositorio, que a la hora de navegarlo se agradece, así si el tester encuentra algo mal puede ir y arreglarlo en la misma rama y no hacer un proceso tan lento como es una pull request

**Reply by laurolmer** [reply]
yo había votado la otra opción pero en realidad ambas me parecen bien

**Comment by ivasansan1** [comment]
Tal como esta el repo ahora mismo, tengo entendido que cuando se cierra una issue la rama asociada también se cierra. Al cerrar la issue del feat para dar paso a los test la rama no se borraría no?

**Reply by laurolmer** [reply]
exactamente

**Reply by marvicmar** [reply]
Claro además lo ideal es que no se realice la pull request hasta que no este la feat completada con sus respectivos test, así seguimos las buenas prácticas, los merge serán más seguros y los action más efectivo.
